app/src/main/java/io/homeassistant/companion/android/widgets/media_player_controls/MediaPlayerControlsWidget.kt

```diff
@@ -392,13 +405,17 @@ class MediaPlayerControlsWidget : BaseWidgetProvider() {
         return AppDatabase.getInstance(context).mediaPlayCtrlWidgetDao().getAll().map { it.id }
     }
 
-    private suspend fun getEntity(context: Context, entityId: String, suggestedEntity: Entity<Map<String, Any>>?): Entity<Map<String, Any>>? {
+    private suspend fun getEntity(context: Context, entityIds: List<String>, suggestedEntity: Entity<Map<String, Any>>?): Entity<Map<String, Any>>? {
         val entity: Entity<Map<String, Any>>?
         try {
-            entity = if (suggestedEntity != null && suggestedEntity.entityId == entityId) {
+            entity = if (suggestedEntity != null && entityIds.contains(suggestedEntity.entityId)) {
```

Comment:
Wouldn't this allow for the widget to switch to for example the second media player that is active when its state is changed, even if the first one is also playing?

The code below always loops though players in the order that they are added, this check uses the latest update instead.

Reply:
From my experience, at least for the media player widget, suggestedEntity seems to always be null. At least i did not find any case where it's not null

Reply:
You're right, it's supposed to pass the entity in `onEntityStateChanged` but it'll always fetch the latest state instead